addons/payment/models/payment_acquirer.py

```diff
@@ -257,8 +286,19 @@ def create(self, values_list):
         return acquirers
 
     def write(self, values):
+        test_acquirers = self.filtered(lambda acq: acq.state == 'test')
+        enabled_acquirers = self.filtered(lambda acq: acq.state == 'enabled')
+
         result = super().write(values)
         self._check_required_if_provider()
+
+        # Handle acquirer disabling.
+        if 'state' in values:
+            if test_acquirers and values['state'] in ('disabled', 'enabled'):
+                test_acquirers._handle_acquirer_disabling()
+            if enabled_acquirers and values['state'] in ('disabled', 'test'):
+                enabled_acquirers._handle_acquirer_disabling()
+
```

Comment:
@AntoineVDV 
I couldn't think of anything better than checking first if the acquirers are in 'test' or 'enabled' and then checking the states they're changed to. I know I could squash the if statements into (probably) one, but I thought for readability, it's better this way.